### PR TITLE
[docs/base] add a type tree for `Number`

### PR DESCRIPTION
**Pull request by inkydragon**
ref: https://github.com/JuliaLang/julia/issues/24741

Generated using the following code with manual modifications

- Sort by type size
- Move `AbstractIrrational` to the end

```jl
using AbstractTrees
AbstractTrees.children(x::Type) = subtypes(x)
print_tree(Number)
```

preview
image

### CODE REVIEW

**Comment by fredrikekre** [comment]
Can you add a blurb before the tree that describes what it shows? Does it make sense to also add an indication which ones are abstract types since the `Abstract*` prefix is not used for all abstract number types?

**Comment by inkydragon** [comment]
Any more comments?

File: doc/src/base/numbers.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
 
 ## Standard Numeric Types
 
+A type tree for all subtypes of `Number`.
```
Comment (oscardssmith) [inline doc/src/base/numbers.md:5]: can this "for number types provided by `Base`"

**Comment by mkitti** [comment]
Should we mark which types are primitive or struct types?

Marking primitives seems noisy. Perhaps we should just mark struct types and indicate the rest are primitive types?

The struct types according to `isstructtype` are:
* BigFloat
* BigInt
* Irrational
* Rational

**Pull request merged**